ui/prometheus-plugin/src/plugins/range-chart-query.ts

```diff
@@ -54,13 +55,32 @@ export function usePrometheusRangeChartQuery(
   const timeRange = useDashboardPrometheusTimeRange();
   const step = usePanelRangeStep(timeRange, minStep);
 
+  // Align the time range so that it's a multiple of the step (TODO: we may
+  // ultimately want to return this from the hook so that charts will know what
+  // time range was actually used?)
+  const { start, end } = useMemoized(() => {
```

Comment:
same question than before, I have the feeling this should be a global hook. I don't really see why it's relative to Prometheus. Probably I'm missing something here. (it was simpler in my head haha)

Reply:
Same thing with this--I see this as Prom-specific behavior at least for now. (And same as above, Grafana treats this as Prom-specific behavior as well and this is in the Prom DataSource.) I think we'll know more when we get to other data sources down the road whether they need/want similar behavior. May not be necessary for them and they may also be dealing in different date/step formats that aren't unix time in seconds.